### DISCUSSION TITLE
Further slider control in the osu!catch editor

### DISCUSSION THREAD

**Discussion by furiner** [ppy/osu#25605] [Ideas]
is it possible to have better control over the curvatures of sliders with their vertices in the osu!catch editor? maps converted from stable have it fine, as displayed here

![osu!_jG4jWLnteH](https://github.com/ppy/osu/assets/30084105/0dd1dcf6-24c5-4121-b869-5eff2b58e0dd)

and it is possible to make the same thing as is but its far more manual and effectively random than it is in stable, and in a lot of cases it ends up looking far uglier and can play differently than expected

![osu!_5j5IRUM5al](https://github.com/ppy/osu/assets/30084105/6c5732f0-1c8f-4e33-9e8b-d31ef175666a)

**Comment by Ryooshuu** [comment]
+1

It would also be nice if we had further-grained control over the sliders, similar to what standard has with bezier and linear points on the sliders.

**Comment by peppy** [comment]
Sounds good. In theory implementation shouldn't be too hard if it's already serialising correctly.

**Comment by Vadeblade** [comment]
To add on this, could the slider kinds be sorta like in Standard? Using Sliders to add curves and split the curve types in the slider where we could get a Linear Slider split into a Perfect curve. Also would like to suggest adding a Slider to stream thing too like in standard mapping would help making streams

**Comment by bdach** [comment]
I'll just mention off hand that this is not as easy as it sounds, because of the way catch beatmaps are basically bastardised osu! beatmaps. To produce a catch beatmap, you take a full osu! beatmap, discard all of the Y coordinates of objects therein, and put the X coordinates on a scrolling playfield where the new "Y coordinate" (if you imagine the entire map as a very long roll of paper that scrolls down/up with time) is moderated by the time.

So we either need a completely new format for catch wherein we don't have to worry about jamming catch juice streams back into osu! slider form, or we need some Hardcore Math to *figure out how to* jam the juice streams back into osu! sliders. Likely something or other involving implicit curves. (Or just do crummy good-enough piecewise-linear approximations instead, but that'll likely lose data.)
